common/static/common/js/discussion/utils.js

```diff
@@ -386,51 +413,55 @@
                         return $3;
                     });
                 } else {
-                    processedText += text;
+                    processedText += tempText;
                     break;
                 }
             }
             /* jshint +W083 */
-            text = processedText;
-            text = text.replace(new RegExp(ESCAPED_DOLLAR, 'g'), '\\$');
-            text = text.replace(/\\\\\\\\/g, ESCAPED_BACKSLASH);
-            text = text.replace(/\\begin\{([a-z]*\*?)\}([\s\S]*?)\\end\{\1\}/img, function($0, $1, $2) {
+            tempText = processedText;
+            tempText = tempText.replace(new RegExp(ESCAPED_DOLLAR, 'g'), '\\$');
+            tempText = tempText.replace(/\\\\\\\\/g, ESCAPED_BACKSLASH);
+            tempText = tempText.replace(/\\begin\{([a-z]*\*?)\}([\s\S]*?)\\end\{\1\}/img, function($0, $1, $2) {
                 return processor(("\\begin{" + $1 + "}") + $2 + ("\\end{" + $1 + "}"));
             });
-            text = text.replace(new RegExp(ESCAPED_BACKSLASH, 'g'), '\\\\\\\\');
-            $div = $("<div>").html(text);
-            cnt = 0;
+            tempText = tempText.replace(new RegExp(ESCAPED_BACKSLASH, 'g'), '\\\\\\\\');
+            $div = edx.HtmlUtils.setHtml($("<div>"), edx.HtmlUtils.HTML(tempText));
             $div.find("code").each(function(index, code) {
-                $(code).html(processor(codeArchive[cnt], 'code'));
-                return cnt += 1;
+                edx.HtmlUtils.setHtml(
+                    $(code),
+                    edx.HtmlUtils.ensureHtml(processor(codeArchive[index], 'code'))
```

Comment:
I see that this would require processor to return an HTML Snippet, or to be wrapped with an HtmlUtil.HTML() call instead.